src/app/events/events.component.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { of } from 'rxjs';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { EventsComponent } from './events.component';
+import { EventsService, TicketmasterEvent } from '../events.service';
+
+describe('EventsComponent', () => {
+  let component: EventsComponent;
+  let fixture: ComponentFixture<EventsComponent>;
+  let eventsServiceSpy: jasmine.SpyObj<EventsService>;
+
+  beforeEach(async () => {
+    eventsServiceSpy = jasmine.createSpyObj('EventsService', ['getEvents']);
+    const mockEvents: TicketmasterEvent[] = [
+      { name: 'Event 1', images: [], url: 'http://example.com/event1' },
```

Comment:
keep irrelevant properties non-specific
```suggestion
      { name: 'foo', images: [], url: 'foo' },
```